libs/entitlement/qa/entitlement-test-plugin/src/main/java/org/elasticsearch/entitlement/qa/test/WritePropertiesCheckActions.java

```diff
@@ -26,6 +27,7 @@ static void setSystemProperty() {
     }
 
     static void clearSystemProperty() {
+        EntitledPlugin.selfTest(); // TODO: find a better home
```

Comment:
This, plus `compileOnly project(":libs:entitlement:qa:entitled-plugin")`, is a little bit cheating; ideally, the ExtensiblePlugin should provide an interface, and the extension would implement it, so they are more decoupled (the common point being just the "API", the interface(s) and set of types these interfaces use).

Reply:
You'd still need a compile-time dependency though, right? I don't see the difference.

Reply:
Interesting. That sounds like the Dependency Inversion Principle applied to modules instead of classes. I'm not sure I see the value of that but it's an intriguing idea.